src/averageAges.js

```diff
@@ -38,6 +45,16 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const motherNames = people.reduce((acc, { mother }) => [...acc, mother], []);
+  let women = people.filter(({ sex }) => sex === 'f');
+
+  withChildren && (
+    women = women.filter(({ name }) => motherNames.includes(name))
+  );
+
+  return women.reduce((acc, { born, died }) => (
+    acc + died - born
+  ), 0) / women.length;
```

Comment:
you do this logic twice, here and inside `calculateMenAverageAge`, a separate function for it would be useful